### PR TITLE
applications: asset_tracker_v2: Enable picolib

### PR DESCRIPTION
**Pull request by simensrostad**
Enable picolib.

### CODE REVIEW

**Review by tejlmand — APPROVED** [review]
lgtm

**Review by jtguggedal — APPROVED** [review]

**Review by jtguggedal — CHANGES_REQUESTED** [review]
Tests should be configured with picolib as well

**Comment by simensrostad** [comment]
@ Confirmed with @nordic-hani that it is safe to enable Picolib. @lemrey If there are no objections concerning nRF Modem Library Ill go a head and get this one merged.

@jtguggedal Application tests has been updated to use Picolib.

**Review by jtguggedal — APPROVED** [review]

**Comment by simensrostad** [comment]
Currently blocked by upstream build warnings

**Comment by lemrey** [comment]
I am not aware of any issues with `picolibc`, if it works for `asset_tracker` there shouldn't be any issues with it.

**Comment by simensrostad** [comment]
> Currently blocked by upstream build warnings

Still valid, will be possible to merge this after next upmerge.

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c
Comment (simensrostad) [reply subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:292]: Please ignore that commit for now. It needs to be increased, but obviously not by that much. Ill add an explanation for it.

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c
Comment (SeppoTakalo) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:295]: What is the point of this?
If the resource or instance is over 999, we end up writing garbage settings.

Why not just return failure or values are over certain limit?

Comment (simensrostad) [reply subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:295]: There are several ways to fix this I'm sure, whatever you prefer really. The buffer can also be increased.
This is the log warning:

```
nd 7 [-Wformat-truncation=]
   43 | #define LWM2M_FIRM_PREFIX "lwm2m:fir"
      |                           ^~~~~~~~~~~
/home/sir1/dev/ncs/zephyr/include/zephyr/sys/printk.h:67:32: note: in definition of macro 'snprintk'
   67 | #define snprintk(...) snprintf(__VA_ARGS__)
      |                                ^~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:294:38: note: in expansion of macro 'LWM2M_FIRM_PREFIX'
  294 |         snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
      |                                      ^~~~~~~~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:1136:52: note: format string is defined here
 1136 |         char path[sizeof(LWM2M_FIRM_PREFIX "65632/0/10")];
      |                                                    ^~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:43:27: note: directive argument in the range [0, 65535]
   43 | #define LWM2M_FIRM_PREFIX "lwm2m:fir"
      |                           ^~~~~~~~~~~
/home/sir1/dev/ncs/zephyr/include/zephyr/sys/printk.h:67:32: note: in definition of macro 'snprintk'
   67 | #define snprintk(...) snprintf(__VA_ARGS__)
      |                                ^~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:294:38: note: in expansion of macro 'LWM2M_FIRM_PREFIX'
  294 |         snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
      |                                      ^~~~~~~~~~~~~~~~~
/home/sir1/dev/ncs/zephyr/include/zephyr/sys/printk.h:67:23: note: 'snprintf' output between 16 and 24 bytes into a destination of size 21
   67 | #define snprintk(...) snprintf(__VA_ARGS__)
      |                       ^~~~~~~~~~~~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:294:9: note: in expansion of macro 'snprintk'
  294 |         snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
```

The problem is that if the integer input variables are large the buffer `path` will not be large enough.

Comment (SeppoTakalo) [reply subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:295]: Add
 ```c
if ((inst < 0 || inst > 9) || (res < 0 || res > 99)) {
	return -EINVAL;
}
```

Before the `sprintf()`. Compiler is clever enough to see now that the values are capped.

Comment (simensrostad) [reply subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:295]: thanks

Comment (simensrostad) [reply subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c]: @SeppoTakalo added

**Comment by simensrostad** [comment]
Depends on https://github.com/nrfconnect/sdk-nrf/pull/10828

File: applications/asset_tracker_v2/sample.yaml
Comment (simensrostad) [inline applications/asset_tracker_v2/sample.yaml:128]: @kacperradoszewski ATv2 + carrier now builds again with the optimizations brought by picolibc. I'm assuming that is why external flash was added to this build?

Comment (kacperradoszewski) [reply applications/asset_tracker_v2/sample.yaml:128]: Yes, this change was to make it fit in CI, fine to remove it if it builds again. Nice that the Thingy build works too 👍

**Review by kacperradoszewski — APPROVED** [review]

**Pull request merged**
